### PR TITLE
Add RGB API support

### PR DESCRIPTION
**Pull request by fatih**
This PR adds support for a new RGB API to allow to print 24-bit colors. Opening for feedback, and will probably merge it as it is. If anyone wants to try it out, check the `TestRGB` test case or the newly added README.md docs.

ss-2024-03-24-113149

### CODE REVIEW

**Review by Integralist — APPROVED** [review]
Great thanks 👍

File: color.go [hunk 1]
```diff
@@ -98,6 +98,9 @@ const (
 	FgMagenta
 	FgCyan
 	FgWhite
+
+	// used internally for 256 and 24bit coloring
```
Comment (bschaatsbergen) [inline color.go:102]:
```suggestion
	// used internally for 256 and 24-bit coloring
```

File: color.go [hunk 2]
```diff
@@ -122,6 +125,9 @@ const (
 	BgMagenta
 	BgCyan
 	BgWhite
+
+	// used internally for 256 and 24bit coloring
```
Comment (bschaatsbergen) [inline color.go:129]:
```suggestion
	// used internally for 256 and 24-bit coloring
```

File: color.go [hunk 3]
```diff
@@ -150,6 +156,30 @@ func New(value ...Attribute) *Color {
 	return c
 }
 
+// RGB returns a newly created foreground 24-bit RGB color.
```
Comment (bschaatsbergen) [inline color.go:159]:
```suggestion
// RGB returns a new foreground color in 24-bit RGB.
```

File: color.go [hunk 4]
```diff
@@ -150,6 +156,30 @@ func New(value ...Attribute) *Color {
...
+func RGB(r, g, b int) *Color {
+	return New(foreground, 2, Attribute(r), Attribute(g), Attribute(b))
+}
+
+// BgRGB returns a newly created 24-bit background RGB color.
```
Comment (bschaatsbergen) [inline color.go:164]:
```suggestion
// BgRGB returns a new background color in 24-bit RGB.
```

File: README.md [hunk 5]
```diff
@@ -30,6 +30,18 @@ color.Magenta("And many others ..")
 
 ```
 
+### RGB colors
+
+If your terminal supports 24-bit colors, you can use RGB color codes
```
Comment (bschaatsbergen) [inline README.md:35]:
```suggestion
If your terminal supports 24-bit colors, you can use RGB color codes.
```

File: color.go [hunk 6]
```diff
@@ -150,6 +156,30 @@ func New(value ...Attribute) *Color {
...
+
+// AddRGB is used to chain foreground RGB SGR parameters. Use as many as parameters to combine
+// and create custom color objects. Example: .Add(34, 0, 12).Add(255, 128, 0).
+func (c *Color) AddRGB(r, g, b int) *Color {
+	c.params = append(c.params, foreground, 2, Attribute(r), Attribute(g), Attribute(b))
+	return c
+}
+
+// AddRGB is used to chain background RGB SGR parameters. Use as many as parameters to combine
+// and create custom color objects. Example: .Add(34, 0, 12).Add(255, 128, 0).
```
Comment (bschaatsbergen) [inline color.go:177]: Reading through the example: `Example: .Add(34, 0, 12).Add(255, 128, 0)`

Is it worth including an example in the README with the ability to chain? I don't think we have that now?

File: color_test.go
Comment (bschaatsbergen) [inline color_test.go:1]: FWIW, but I can do it in another PR, we should still include a test that covers the chaining. What do you think?

**Review by bschaatsbergen — APPROVED** [review]
Couple nitpicks on the comments but, LGTM 🚀 
API seems straightforward and easy to use. It's a great addition to the existing supported ANSI color palette.

Thanks for picking this up 👍🏼

**Comment by fatih** [comment]
Thanks for to the reviews. I want to keep this open for some time to try it out before I merge it.

**Comment by artificial-aidan** [comment]
Another vote of confidence. Worked great

**Comment by bschaatsbergen** [comment]
Hey @fatih, did you have a chance to take a look at this PR?

**Comment by fatih** [comment]
@bschaatsbergen yeap, I'll merge this soon. Hadn't time go over it.

**Comment by amterp** [comment]
This change will be great! Any chance we can get it released soon-ish?

**Comment by fatih** [comment]
@amterp thanks for the ping, I'll cut a release today.

**Comment by fatih** [comment]
Here is the release: https://github.com/fatih/color/releases/tag/v1.18.0

**Comment by amterp** [comment]
Looks good! 😄 Thanks @fatih 

image

**Pull request merged**